packages/image_picker/image_picker_for_web/lib/image_picker_for_web.dart

```diff
@@ -37,9 +37,63 @@ class ImagePickerPlugin extends ImagePickerPlatform {
     double maxHeight,
     int imageQuality,
     CameraDevice preferredCameraDevice = CameraDevice.rear,
-  }) {
+  }) async {
     String capture = computeCaptureAttribute(source, preferredCameraDevice);
-    return pickFile(accept: _kAcceptImageMimeType, capture: capture);
+    return pickFile(
+      accept: _kAcceptImageMimeType,
+      capture: await _resizeImage(
+        capture,
+        maxWidth.toInt(),
+        maxHeight.toInt(),
+        imageQuality,
+      ),
+    );
+  }
+
+  static Future<String> _resizeImage(
+    String src,
+    int maxWidth,
+    int maxHeight,
+    int imageQuality,
+  ) {
+    final completer = Completer<String>();
+    final img = html.ImageElement();
+    img.onError.listen((event) {
+      completer.complete("");
+    });
+    img.onLoad.listen((event) {
+      final canvas = html.CanvasElement();
+      final ctx = canvas.context2D;
+      double ratio = 1;
+
+      if (img.width > img.height) {
+        if (img.width > maxWidth) ratio = maxWidth / img.width;
+      } else if (img.height > maxWidth) {
+        ratio = maxWidth / img.height;
+      }
+      if (img.width > img.height) {
+        if (img.width < maxHeight) ratio = maxHeight / img.width;
+      } else if (img.height < maxHeight) {
+        ratio = maxHeight / img.height;
+      }
+
+      canvas.height = (img.height * ratio).floor();
+      canvas.width = (img.width * ratio).floor();
+      // Draw the image to canvas and resize
+      ctx.drawImageScaled(img, 0, 0, canvas.width, canvas.height);
+      final base64 = canvas.toDataUrl("image/png", imageQuality / 100);
+      completer.complete(base64);
+    });
+    img.src = src;
+    // make sure the load event fires for cached images too
+    if (img.complete) {
+      // Flush cache
+      img.src =
+          'data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///ywAAAAAAQABAAACAUwAOw==';
```

Comment:
Will this trigger the onLoad event above? If so, you might want to check for this special image to do nothing with it; otherwise you'll attempt to complete the `completer` several times, and it'll throw an exception (you can only complete a completer once)

Reply:
Yes it's a bit weird here and I don't really understand what's going on.
Maybe @postacik could help us here (?).

Reply:
@ditman, you are right, if it does trigger the onload event several times, it would cause exceptions. The JavaScript code I used was not using a Promise (or Completer) but simply copying the result to the canvas so the hack to avoid cases where onload did not fire did not have any side effects but here it would. The base64 image is a 1x1 gif. Maybe the completer may be avoided if the loaded image size is 1x1.

Reply:
Added a simple `img.width > 1 && img.height > 1` check before performing the resize logic.
Give me your feedback here!